crypto/asn1/tasn_enc.c

```diff
@@ -383,7 +383,13 @@ typedef struct {
     const ASN1_VALUE *field;
 } DER_ENC;
 
-static int __cdecl der_cmp(const void *a, const void *b)
+static int
+#ifdef __GNUC__
+  __attribute__((cdecl))
```

Comment:
Is the GNU bit necessary?

Reply:
Is this only valid for 32 bit?

Reply:
Also I think there are multiple qsort calls in openssl (at least in master currently this appears to be the case).
So it might need to be included from an internal header.

Reply:
> Is the GNU bit necessary?

At least Clang support using MS CRT and custom calling convetions.

> Is this only valid for 32 bit?

No, even on 64bit Windows has multiple calling conventions.

> Also I think there are multiple qsort calls in openssl (at least in master currently this appears to be the case).
> So it might need to be included from an internal header.

When I tried compiling I only got an error here, but it's possible more places would only be reported if I tried compiling with the patch.

Reply:
> even on 64bit Windows has multiple calling conventions

So should this only be on Windows?

On my local Ubuntu, the `qsort()` comparison function is `typedef`'d as

```
typedef int (*__compar_fn_t) (const void *, const void *);
```

i.e. with no calling convention overrides.

Maybe it would be better to `#define` a `QSORT_COMPAR_DECOR` to `__cdecl` (or whatever) for Windows, and depending on compiler etc, and then

```
#ifndef QSORT_COMPAR_DECOR
#define QSORT_COMPAR_DECOR
#endif

static int QSORT_COMPAR_DECOR der_cmp(const void *a, const void *b)
```